Better onboarding on demo instance

fm3 opened the pull request:
Lots of improvements for new users on webknossos.org:
- Signup on frontpage
- 1 organization per new user
- Public Datasets hidden in list (only on demo)
- Show featured publications in dashboard
- New user banner with information
- Better welcome email
- Improved info tab in annotation view
- Renamed tracing -> annotation
- Improved docs
- Refactored DatasetCache from Dashboard in own component with context

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~
- [x] Updated [documentation](../blob/master/docs) if applicable
- ~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~
- ~[ ] Needs datastore update after deployment~
- [x] Ready for review

philippotto commented on frontend/javascripts/oxalis/view/version_view.js:
```diff
@@ -70,9 +70,9 @@ class VersionView extends React.Component<Props, State> {
-                  You are currently previewing older versions of this tracing. Either restore a
-                  version by selecting it or close this view to continue tracing. The shown tracing
-                  is in <b>read-only</b> mode as long as this view is opened.
+                  You are currently previewing older versions of this annotation_actions. Either
```
```suggestion
                  You are currently previewing older versions of this annotation. Either
```

Probably?

normanrz commented on frontend/javascripts/dashboard/dataset/dataset_cache_provider.js:
```diff
@@ -0,0 +1,116 @@
+  }
+  useEffect(() => {
+    try {
+      persistence.load(history);
```
I think I resolved this issue, please check.

philippotto commented on frontend/javascripts/dashboard/dataset_view.js:
```diff
@@ -188,105 +131,108 @@ class DatasetView extends React.PureComponent<Props, State> {
+  function renderTable() {
+    const filteredDatasets = features().isDemoInstance
+      ? context.datasets.filter(
+          d => !d.isPublic || d.owningOrganization === props.user.organization,
```
Is this correct? On the demo instance, the user is only supposed to see datasets which are either **not** public (?) OR which belong to their own organization? Shouldn't it be `d.isPublic`?

normanrz replied:
This is a bit weird, I agree. The problem I was trying to solve is that when a new user arrives on wk.org, they are presented with a massive amount of non-descriptively labeled datasets of which none (probably) are their own. This hack was to filter all published datasets from other orgs out. They would still be available through the publication tab, tho.
Not sure what the best way to handle this would be.

philippotto replied:
Ok, but wouldn't this be equivalent to the following?

```
d => d.owningOrganization === props.user.organization
```

The current boolean expression could render datasets which are not public and belong to a different organization, but I assume the back-end wouldn't send these datasets in the first place?

normanrz replied:
Yes, you are right :D

philippotto commented on frontend/javascripts/dashboard/registration_form.js:
```diff
@@ -0,0 +1,178 @@
+
+type Props = {|
+  form: Object,
+  onRegistered: true => void,
```
I think, the parameter `true` of the `onRegistered` function is never used (otherwise, it probably should be a `boolean`).

normanrz replied:
It is expected by the calling code. This has todo with the shared API of both registration forms.

normanrz replied:
For this version of the registration form, it is always `true` therefore the literal type.

philippotto commented on frontend/javascripts/dashboard/registration_form.js:
```diff
@@ -0,0 +1,178 @@
+  return output;
+}
+
+class RegistrationForm extends React.PureComponent<Props, State> {
```
Do I understand this right, that this component is an adapted copy of the "old" `RegistrationForm`, right? Are the difference to big to unify these two components into one? If not, I'd rename this component to something like `AdminRegistrationForm` to avoid confusion with the other RegistrationForm.

normanrz replied:
I think both forms have diverged a bit and it would make sense to split them into 2 components. Renaming is a good idea, tho.

normanrz replied:
renamed to `SpotlightRegistrationForm` in latest commit.

normanrz commented:
@fm3 @youri-k I think it would be nice to add the organization's display name to `POST /api/auth/createOrganizationWithAdmin`. And a route `GET /api/organizations/:organizationName` to resolve the organization's display name.

fm3 commented:
@normanrz I added the organizationDisplayName to the query as specified. if `organization` is emptystring, the backend will use a normalized version of organizationDisplayName for the name.
Also, I added `/organizations/:organizationName` as you requested as well as restricting `/organizations` so that it will fail if there are multiple organizations on the server. I did not yet adapt the frontend to those changes (except for setting some values to default values so that the requests don’t fail)

normanrz commented:
@fm3 For deployments with a single org, can we have a route like `/api/organizations/default` that returns the single org?

Also, the "organziation team" uses the cryptic org name, can we change that to something more generic like "Default"?

philippotto commented:
What's the status here?

normanrz commented:
I think the code should be ready, but I need to do one more round of checking.

fm3 commented:
I didn’t do the frontend changes for switching to /organizations/default, I guess those are still to do?

philippotto commented:
@fm3 I just merged the master into this PR and resolved most of the conflicts. However, there are still conflicts in `app/controllers/Authentication.scala` which I didn't feel comfortable to resolve. Could you have a look at these if you have a minute? :)

philippotto commented on conf/webknossos.latest.routes:
```diff
@@ -169,7 +170,9 @@ GET           /statistics/webknossos
 GET           /statistics/users                                                 controllers.StatisticsController.users(interval: String, start: Option[Long], end: Option[Long], limit: Int)
 
 #Organizations
-GET           /organizations                                                    controllers.OrganizationController.listAllOrganizations
```
@fm3 The onboarding process still contains the possibility to join an existing organization, which is why this route is still needed. Or is the front-end part out of date and it shouldn't be possible anymore to follow the "onboarding steps" to join an existing organization?

fm3 replied:
I think the front-end route part is out of date. The idea was that listing all organizations could be considered a data leak, so joining existing organizations should be possible only via invite link, meaning that this list request should be obsolete.

philippotto commented:
@fm3 Sorry to bug you again, but the `getDefaultOrganization` route returns an error message (à la "cannot list organizations due to privacy reasons"), but the front-end cannot know whether this is an multi-orga setup or not (at least, I don't think so). Could you just return `null` if there are multiple organizations?
